core/base/src/TROOT.cxx

```diff
@@ -526,8 +526,23 @@ namespace Internal {
       return macroPath;
    }
 
+   // clang-format off
    ////////////////////////////////////////////////////////////////////////////////
    /// Enables the global mutex to make ROOT thread safe/aware.
+   ///
+   /// The following becomes safe:
+   /// - concurrent construction and destruction of TObjects, including the ones registered in ROOT's global lists (e.g. gROOT->GetListOfCleanups(), gROOT->GetListOfFiles())
+   /// - concurrent usage of _different_ ROOT objects from different threads, including ones with global state (e.g. TFile, TTree, TChain) with the exception of graphics classes (e.g. TCanvas)
+   /// - concurrent calls to ROOT's type system classes, e.g. TClass and TEnum
+   /// - concurrent calls to the interpreter through gInterpreter
+   /// - concurrent loading of ROOT plug-ins
+   ///
+   /// In addition, gDirectory, gFile and gPad become a thread-local variable which defaults to a TDirectory singleton.
```

Comment:
> defaults to a TDirectory singleton.

it is not an 'unamed' singleton.  It defaults to gROOT.

Reply:
I'm confused, `gROOT` is not a `TDirectory`..?

Reply:
It is :) so your sentence is correct and accurate :).  But gROOT is also 'well known' so why not mention 'which'  TDirectory singleton is being used :)

Reply:
oh damn `TROOT` inherits from `TDirectory`. my bad.

Reply:
wait...`gDirectory`, `gFile` and `gPad` all default to `gROOT`?

Reply:
no, just gDirectory ....

checking what gFile and gPad default to ...

Reply:
the other default to nullptr (if I read the code correctly).

Reply:
ok thanks, fixed